packages/nimble-components/build/generate-icons/source/index.js

```diff
@@ -2,10 +2,12 @@
  * Build script for generating Nimble components for icons.
  *
  * Iterates through icons provided by nimble-tokens, and generates a Nimble component for each in
- * src/icons. Also generates an all-icons barrel file.
+ * src/icons. Also generates an all-icons barrel file and the icons.mdx Storybook file.
  */
 import { pascalCase, spinalCase } from '@microsoft/fast-web-utilities';
 import * as icons from '@ni/nimble-tokens/dist/icons/js';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import { iconMetadata } from '@ni/nimble-components/dist/esm/icon-base/icon-metadata';
```

Comment:
There is a chicken and egg problem here. The components build needs the icons to be generated before typescript runs to create the dist/esm output. The icons need the dist/esm output to generate the story.

Some options:
- Have a separate script that runs after components build but before storybook build to generate the icon story
- Can we just make an icons story in mdx that loops over all the available icons? I'd expect we can do something like that in the React mdx syntax

Reply:
Maybe something like this comment could work? https://github.com/storybookjs/storybook/issues/8392#issuecomment-660851718